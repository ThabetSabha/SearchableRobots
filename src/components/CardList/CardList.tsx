import React from "react";
import Card from "../Card/Card";

const CardList = ({ robots }: CardListProps) => {
  return (
    <>
      {robots.map((user) => {
        return (
          <Card
            key={user.id}
            id={user.id}
            name={user.name}
            email={user.email}
          />
        );
      })}
    </>
  );
};

export default CardList;
