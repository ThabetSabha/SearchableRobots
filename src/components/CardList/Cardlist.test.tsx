import React from "react";
import CardList from "./CardList";
import { shallow } from "enzyme";

test("That the Card Component Renders", () => {
  let cardListProps: CardListProps = {
    robots: [{ id: 1, name: "Test", email: "Test@gmail.com" }],
  };

  let wrapper = shallow(<CardList {...cardListProps} />);
  expect(wrapper).toMatchSnapshot();
});
