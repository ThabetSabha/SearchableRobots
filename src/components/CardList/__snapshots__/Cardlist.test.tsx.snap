// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`That the Card Component Renders 1`] = `
<Fragment>
  <Card
    email="Test@gmail.com"
    id={1}
    key="1"
    name="Test"
  />
</Fragment>
`;
